src/main/java/rx/internal/operators/OperatorZip.java

```diff
@@ -227,7 +224,7 @@ void tick() {
                 // nothing yet to do (initial request from Producer)
                 return;
             }
-            if (COUNTER_UPDATER.getAndIncrement(this) == 0) {
+            if (counter.getAndIncrement() == 0) {
```

Comment:
`counter` should be read into a local variable before the loop if not moved into `Zip`'s class hierarchy.